Module1 Homework/README.md

```diff
@@ -0,0 +1,184 @@
+
+- Downloading data from a remote repository:
+
+    - `git clone` : creates a local repository from a remote repository 
+    - `git fetch` : fetches any changes from a remote repository but does not merge these changes into the local repositories working directory (ie. we can see that there are changes on the server, but they haven’t impacted the local working code).
+    - `git pull` : fetches changes from a remote repository and merges those changes into the local repositories branch.
+
+- Sending data to a remote repository:
+    - `git push`: sends the local staged and committed files to the remote repository 
+
+#### _s) Why is a programming To-Do list important, and what is a minimum viable product?_
+
+A `To-Do list` helps the programmer to think about the functionality of the program upfront. This helps to define what the program should achieve before actually implementing the functionality, aiding in clarifying the goals of the program before development begins.
+
+A `minimum viable product` (MVP) is an app that includes just the core functionality required to make the program perform its vital ("must-haves") functions.
+
+#### _t) What is a simple way of describing Binding in SwiftUI?_
+
+A bidirectional interface between a SwiftUI component the user interacts with (e.g., a Slider) and the underlying stored value (e.g., a Float). In the slider example: a change of the Slider by the user automatically updates the underlying Float value it represents. Likewise, a change in the stored Float value automatically updates the position of the Slider seen by the user.
+
+One very nice way that I have heard a Binding described is: "a reactive glue between a UI component and a data source - holding them together so that changes in one are automatically reflected in the other."
+
+#### _u) What command do you use in git to move changes from one branch to another? (Feel free to answer with how you use git, i.e. terminal or another app)_
+
+`git merge`
+
+For example if you wanted to move the changes from branch `feature-branch` into `main`:
+
+- Ensure you are on the `main` branch 
+    - if not, switch to the `main` branch using: `git checkout main`.
+- Merge in the changes from `feature-branch` using: `git merge feature1`.
+
+### _v) What is the type of the variable defined below?_
+``` 
+var a = 87
+```
+
+`Int` (integer)
+
+#### _w) What is the difference between var and let?_
+
+- A `var` is a variable, meaning that the value assigned to it can be changed over the lifetime of its scope.
+- A `let` is a constant, meaning that the value assigned to it is fixed and cannot be changed over the lifetime of its scope.
+
+
+## Part 2 - Programming Assignment
+
+
+Refer to the code in `ContentView.swift`.
+
+
+## Part 3 - Above and Beyond
+
+> These answers refer to code in the Homework document.
+
+In ContentView, lines 1 and 2 show the definition of `State Properties`.
+
+In ContentView, line 3 shows the definition of an `Instance Property`.
```

Comment:
Instance is okay, however we're looking here more towards stored property. This is the important difference between lines 3 and 4. One is a stored property while the other is computed.

Reply:
True.  Stored property is much more Swifty sounding (especially for a struct).